From 13d96f49c7dffaa176b2bb93763aa9adb03ac4d7 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Tue, 5 Aug 2025 17:08:31 +0800
Subject: [PATCH 224/236] mtk: hostapd: fix acs + ht coex issue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When all radios have ACS enabled and 2G has HT Coex enabled,
the scanning process will proceed as follows:
2G ACS → 2G HT SCAN → 5G ACS → 6G ACS.

However, the current logic for performing sequential ACS only
triggers the next ACS when the hapd state of the previously
scanned interface equals HAPD_IFACE_ACS.
When 2G HT Coex is enabled along with ACS, the state becomes
HAPD_IFACE_HT_SCAN after the HT SCAN.
As a result, the ACS for the subsequent radio will not be triggered.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 src/ap/hostapd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index e3eab9066..22b1b80ff 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -3038,7 +3038,7 @@ int hostapd_setup_interface_complete(struct hostapd_iface *iface, int err)
 	unsigned int i;
 	int not_ready_in_sync_ifaces = 0;
 
-	if (iface->state == HAPD_IFACE_ACS) {
+	if (iface->state == HAPD_IFACE_ACS || iface->state == HAPD_IFACE_HT_SCAN) {
 		int i;
 
 		for (i = 0; i < interfaces->count; i++) {
-- 
2.45.2

